### DISCUSSION TITLE
Please Support for CodeFormer

### DISCUSSION THREAD

**Discussion by mo95** [hlky/stable-diffusion-webui#1228] [Feature suggestions]
In my testing, [CodeFormer](https://github.com/sczhou/CodeFormer) works much better than GFPGAN, it shows great robustness with better facial details and sharper enhanced results, see [examples](https://replicate.com/sczhou/codeformer/examples) provided by authors. For me, most results of GFPGAN v3 are over smooth. 

Hope this feature of CodeFormer could be added in the future.

**Comment by AdrianTheFrog** [comment]
I agree that it should be added, but it should not replace GFPGAN because (from my quick testing on the huggingface demo) it works well for upscaling but does not seem to do that good of a job at fixing mangled faces.

EDIT: on the documentation it says that codeformer support is planned (sd-webui.github.io/stable-diffusion-webui/docs/5.upscalers.html)
